### PR TITLE
bugfix/AB#88157_SUI-clicking-in-a-point-of-an-expanded-cluster-should-open-the-country-based-on-the-latitude-longitude-of-the-point

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

- feat: check if in cluster layers click event it's propagated from a child marker or just default layer, in the first case we will use the lat lng from the propagated marker, otherwise default event would be taken. This method it's easier than adding click events to all child markers, as we are handling just one event listener per marker groups, and there would be no conflict between cluster click or child marker click

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/88157)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please refer to video below

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/123092672/ed4523e7-d878-4c85-9271-2e602ef7764a

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
nice, will test that right away!

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp nice, will test that right away!

Taking a look at this, a very good improvement would be to trigger the country fetch if current selected country is different from the previous one, because clicking on each feature is triggering same request 😕 

Still need to update some more things @AntoineRelief , please don't merge it yet, as it's working when cluster markers are expanded like that, but if user zooms in and markers are place individually, clicking on them won't work

I'll update full fix soon

File: libs/shared/src/lib/components/ui/map/layer.ts [hunk 1]
```diff
@@ -874,8 +874,17 @@ export class Layer implements LayerModel {
...
     }
     this.rulesListener = (e) => {
+      let event: any = e;
+      if (
+        get(this.layerDefinition, 'featureReduction.type') === 'cluster' &&
+        // If it's actually one of the features within the cluster, then take that one
+        e.propagatedFrom.feature
+      ) {
+        const { latitude, longitude } = e.propagatedFrom.feature.properties;
+        event = { latlng: { lat: latitude, lng: longitude } };
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/ui/map/layer.ts:891]: @unai-reliefapp 
I'll fix it, but there is something wrong therre
indeed, your logic is based on the fact that the feature would have latitude & longitude properties, but it's not always the case
indeed, the lat & long are coming from 2 fields that the admin can select
it's lucky that for the HR layer you tested they are called latitude longitude, but there is another layer with different fields, and it won't work for this other layer
better use a consistent field like feature.geometry.coordinates

Comment (unai-reliefapp) [reply libs/shared/src/lib/components/ui/map/layer.ts:891]: > @unai-reliefapp I'll fix it, but there is something wrong therre indeed, your logic is based on the fact that the feature would have latitude & longitude properties, but it's not always the case indeed, the lat & long are coming from 2 fields that the admin can select it's lucky that for the HR layer you tested they are called latitude longitude, but there is another layer with different fields, and it won't work for this other layer better use a consistent field like feature.geometry.coordinates

Please check my last comment, i'll push the fix, it's not indeed a good fix

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
I just pushed one

if you have anything else to push please let me know

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp I just pushed one
> 
> if you have anything else to push please let me know

It's fine, yes, idk the markers in the cluster where not triggering the rule event on clicking on them, fine for me 👍 
thanks

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp I just pushed one
> 
> if you have anything else to push please let me know

yes, if you zoom in the map and a cluster separates into it's markers, if you click on one, there is no event triggered because the event it's only placed in the cluster

**Pull request merged**